server/src/routes/api/v1/podcastVotesRouter.js

```diff
@@ -0,0 +1,105 @@
+import express from 'express'
+import objection from 'objection'
+const { ValidationError } = objection
+
+import { Vote } from '../../../models/index.js'
+
+const podcastVotesRouter = new express.Router({ mergeParams: true })
+
+podcastVotesRouter.get('/', async (req, res) => {
+  const { podcastId } = req.params
+  let userId
+  if (req.user) userId = req.user.id
+
+  try {
+    let votes = {}
+    votes.userVotes = await Vote.query()
+      .where('podcastId', podcastId)
+      .andWhere('userId', userId)
+      .first()
+    votes.totalVotes = await Vote.query()
+      .where('podcastId', podcastId)
+      .sum('value as value')
+      .first()
+    return res.status(201).json({ votes: votes })
+  } catch (error) {
+    if (error instanceof ValidationError) {
+      return res.status(422).json({ errors: error.data })
+    }
+    return res.status(500).json({ errors: error })
+  }
+})
+
+podcastVotesRouter.post('/', async (req, res) => {
+  const { value } = req.body
+  const { podcastId } = req.params
+  const userId = req.user.id
+
+  try {
+    let newVote = {}
+    newVote.userVotes = await Vote.query()
+      .insert({ podcastId, userId, value })
+      .first()
+      .returning('*')
```

Comment:
wait...why are we calling first here? I believe you should delete line 42
https://vincit.github.io/objection.js/recipes/returning-tricks.html
you only need `first` if you have a where statement generally.

Ill allow line 43, since its best practice with PG databases, though this will break with any other database. generally though, we want to stick to one pattern. So if you have `insertAndFetch` elsewhere in your codebase (which is the alternative to using `returning`) you would want to replace those queries with insert queries with `returning`

Reply:
`userVotes` is both a poor name (since it should contain but one vote) and should exist on its own, not as a property of `newVote`. We don'y always need to assign our data to properties on one object to be passed back